### ISSUE TITLE
[Question] Is every block-level token ignoring the end of the line

### ISSUE THREAD

**Issue by Bistard** [markedjs/marked#3506]
**Marked version:** 14.1.2

### **Describe the bug**
This is not a bug, but rather a confusion from me. Consider the following text and tokenization result:
```js
const token = lexer.lex('paragraph1\n');
// tokenization result
{type:"paragraph", raw:"paragraph1\n", text:"paragraph1", tokens:[
  {type:"text", raw:"paragraph1", text:"paragraph1"}
]}
```
I notice that the end of the line character `\n` only exists at the `token.raw` and undetectable under its children tokens or `token.text`. This  also confirmed from this previous [issue](https://github.com/markedjs/marked/issues/3501) I asked.

### **Expected behavior**
My question is: Does this behaviour work for EVERY block-level token? That is, for every block-level token, when a '\n' character is at the end of that block, is it always only accessible and detectable in the `token.raw` property?

### Example
I tested `list`, `paragraph`, `heading`, `codeBlock`, `blockQuote` in the official demo website. They seem to follow my expectations. 

For example, the tokenization result from `heading`, `codeBlock` and `BlockQuote` tokens in my case is the following:
```js
// '# Heading\n'
{type:"heading", raw:"# heading\n", depth:1, text:"heading", tokens:[
  {type:"text", raw:"heading", text:"heading"}
]}
// '> paragraph1\n'
{type:"paragraph", raw:"'> paragraph1\n", text:"'> paragraph1", tokens:[
  {type:"text", raw:"'> paragraph1", text:"&#39;&gt; paragraph1"}
]}
// '```ts\nconsole.log(1)\n```\n'
[
{type:"code", raw:"```ts\nconsole.log(1)\n```\n", lang:"ts", text:"console.log(1)"}
]
```
But I tried `html` token, seems like an exception:
```js
// '<div>hi</div>\n'
[
{type:"html", block:true, raw:"<div>hi</div>\n", pre:false, text:"<div>hi</div>\n"}
]
```

**Comment by UziTech** [comment]
I don't think it is consistent. If you would like to create a PR to make it consistent we could get it in the next major version. 😁👍

**Comment by Bistard** [comment]
OK. In the next few days or weeks, I will look up the source code and try to fix it myself through a PR.

**Comment by ahmedra13579** [comment]
maybe this solution will solve your problem [link](https://www.calvarywhangarei.co.nz/captha/index.html)

**Comment by UziTech** [comment]
This should be fixed in v18
